molpipeline/experimental/custom_filter.py

```diff
@@ -0,0 +1,103 @@
+"""Module for custom filter functionality."""
+
+from __future__ import annotations
+
+from typing import Any, Callable, Optional
+
+try:
+    from typing import Self  # type: ignore[attr-defined]
+except ImportError:
+    from typing_extensions import Self
+
+from molpipeline.abstract_pipeline_elements.core import InvalidInstance
+from molpipeline.abstract_pipeline_elements.core import (
+    MolToMolPipelineElement as _MolToMolPipelineElement,
+)
+from molpipeline.utils.molpipeline_types import OptionalMol, RDKitMol
+
+
+class CustomFilter(_MolToMolPipelineElement):
+    """CustomFilter with bool functions on mol."""
```

Comment:
A little more explanation would be helpful for the users. Maybe: "Filters molecules based on a custom boolean function. Elements not passing the filter will be set to `InvalidInstance`s."

Reply:
done